### ISSUE TITLE
Can't load persp-mode

### ISSUE THREAD

**Issue by bennigoetz** [syl20bnr/spacemacs#5100]
Minimal Emacs experience, new to Spacemacs.

I installed Spacemacs on top of Emacs 24.5.1, and got the error listed below. Running emacs --insecure didn't make any difference to the error. I have no idea what else to try to resolve the issue.

Otherwise Spacemacs seems to operate fine, but having this error and extra buffer pop up every time is annoying.

```
Error (use-package): persp-mode :init: Symbol's function definition is void: persp-mode
```

**Comment by StreakyCobra** [comment]
I suppose you are on windows? Can you check you have `git` and `tar` installed and that both are in `exec-path` (`M-: exec-path RET`)?

**Comment by bennigoetz** [comment]
Thanks for the quick reply!

I'm actually on CentOS 6.7, running on the Stampede cluster at TACC. git
and tar are both installed on the system, and are available on my system
path (I double-checked by running them at a prompt).

Benni

**Comment by StreakyCobra** [comment]
Being on the system path and accessible from a terminal doesn't mean Spacemacs can find it. Can you check within emacs with `M-: exec-path RET`? Also, can you try by adding `exec-path-from-shell` to `dotspacemacs-excluded-packages` in your `.spacemacs`?

**Comment by bennigoetz** [comment]
OK, I think I did as you suggested.

`which git` returns `/usr/bin/git`, and `/usr/bin` is in `exec-path`.
`which tar` returns `/bin/tar`, and `/bin` is in `exec-path`.

Forgive my ignorance of elisp, I hope this was correct:
`dotspacemacs-excluded-packages '()` to `dotspacemacs-excluded-packages '(exec-path-from-shell)`

Same error message about `persp-mode` after that adjustment. All the other packages downloaded just fine after I first installed Spacemacs. (Actually, I think three of them had errors, but all but `persp-mode` installed after running `emacs --insecure`.)

**Comment by StreakyCobra** [comment]
All what you did seems correct, I'm clueless then. Just to be sure can you try again with `exec-path-from-shell` excluded, but before starting Spacemacs deleting these two folders:

- `.emacs.d/elpa/persp-mode-[version]`
- `.emacs.d/.cache/quelpa`

**Comment by bennigoetz** [comment]
Just tried what you suggested, but I couldn't delete

`.emacs.d/elpa/persp-mode-[version]`

because it wasn't in the `.emacs.d/elpa` directory. I don't know whether that's useful information.

Same error after starting up Spacemacs.

Sounds like this is a rare bug, of mysterious origin. Perhaps it will just as mysteriously vanish with 105.10. Thanks for the help!

**Comment by StreakyCobra** [comment]
So I really don't know, sorry :-(

**Comment by syl20bnr** [comment]
Since you did not have any `persp-mode` folder in the `elpa` directory it seems to be an issue with Git because this particular package is cloned from github directly via quelpa.

What is your git version and tar version ? CentOS 6 should have pretty old packages.

**Comment by bennigoetz** [comment]
My `git` version is 1.7.1, and my `tar` version is 1.23. So yeah, they're both pretty old. I'll try installing new versions in my own bin directory, and see if that fixes things. Unfortunately it'll have to wait a day or two to test it out, as I have to prepare for a lecture tomorrow.

BTW, awesome project, I'm very happy to see this approach. And I'm floored by all the personal attention for such a small problem. Thank you!

**Comment by tyuownu** [comment]
For me, I just copy the persp-mode to elpa/ directory, it solve the problem.

**Comment by warreq** [comment]
Just chiming in to say I'm also experiencing this issue with persp-mode.

```
git --version
git version 2.7.1
```
```
tar --version
tar (GNU tar) 1.28
```

persp-mode folder doesn't show up under elpa/, and the output from emacs every startup outputs:

```
(Spacemacs) --> installing package: persp-mode@spacemacs-layouts... [1/1]
Fetcher: github
Source: syl20bnr/persp-mode.el

Updating /home/man/.emacs.d/.cache/quelpa/build/persp-mode/
Error (use-package): persp-mode :init: Symbol's function definition is void: persp-mode
```

I tried clearing out the .cache folder but that did not change anything.

**Comment by bennigoetz** [comment]
Huh. So I went to the [persp-mode Github page](https://github.com/Bad-ptr/persp-mode.el), and followed the directions to install persp-mode. Specifically, I ran:
`M-x: package-install RET persp-mode RET`

That seems to have fixed the problem. After quitting Spacemacs and then opening it again, no warning, and I see the persp-mode directory in the `.emacs.d/elpa` directory.

**Comment by KanTian** [comment]
thx @bennigoetz, I follow your instructions and it works!
I install persp-mode with the second method :

> Alternatively you can download the persp-mode.el from [persp-mode Github page](https://github.com/Bad-ptr/persp-mode.el), and install it as a package:
> `M-x: package-install-file RET 'path_to_where_you_saved_persp-mode.el' RET`

**Comment by remd** [comment]
I'm also experiencing this problem after installing spacemacs on Windows 7 64-bit over the `emacs-24.5-bin-i686-mingw32` prebuilt binaries.

I am using `runemacs.exe --insecure` or `emacs.exe --insecure` and in the minibuffer I get the following error message when following @bennigoetz instructions to manually install persp-mode.

`Could not create connection melpa.org:443`

There's also some minibuffer output before that indicates it's using TLS to connect. I'm not sure how to save minibuffer output or I would paste it here.

**Comment by bmag** [comment]
@remd just for the record, I recommend the 64bit binaries of Emacs from https://sourceforge.net/projects/emacsbinw64/files/. It is compiled with GnuTLS and image support built-in.

**Comment by remd** [comment]
@bmag I was able to install all other packages [121/122] other than persp-mode under `emacs --insecure` on the binaries I was using previously. I'm now using the 64-bit build you linked and everything installed properly the first time. Thanks!

**Comment by braineo** [comment]
@bennigoetz  Thank you that solves the problem!

**Comment by aBathologist** [comment]
A tip for those who might run into a similar hangup: On Windows 7, I tried adding `git` to my `Path` variable. While I could then run `git` from the Windows shell, but, *when I launched the application via file explorer*, Emacs still wasn't installing `persp-mode` on startup. When I launched emacs from the shell, it registered the update to the `Path` variable and installed `persp-mode` as expected! Thanks, to all for working this out.
